src/main/java/com/wanted/babdoduk/restaurant/domain/review/service/ReviewService.java

```diff
@@ -0,0 +1,90 @@
+package com.wanted.babdoduk.restaurant.domain.review.service;
+
+import com.wanted.babdoduk.restaurant.domain.restaurant.repository.RestaurantRepository;
+import com.wanted.babdoduk.restaurant.domain.review.dto.RestaurantReviewRequestDto;
+import com.wanted.babdoduk.restaurant.domain.review.dto.RestaurantReviewResponseDto;
+import com.wanted.babdoduk.restaurant.domain.review.entity.RestaurantReview;
+import com.wanted.babdoduk.restaurant.domain.review.entity.RestaurantReviewScore;
+import com.wanted.babdoduk.restaurant.domain.review.exception.ReviewNotFoundException;
+import com.wanted.babdoduk.restaurant.domain.review.repository.RestaurantReviewRepository;
+import com.wanted.babdoduk.restaurant.exception.NotFoundRestaurantException;
+import com.wanted.babdoduk.user.exception.UserNotFoundException;
+import com.wanted.babdoduk.user.domain.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ReviewService {
+
+    private final RestaurantReviewRepository reviewRepository;
+    private final UserRepository userRepository;
+    private final RestaurantRepository restaurantRepository;
+
+    @Transactional
+    public RestaurantReviewResponseDto createRestaurantReview(
+            Long userId,
+            Long restaurantId,
+            RestaurantReviewRequestDto reviewRequestDto) {
+
+        checkUserExists(userId);
+
+        if (!restaurantRepository.existsById(restaurantId)) {
+            throw new NotFoundRestaurantException();
+        }
+
+        return toReviewResponseDto(reviewRepository.save(
+                RestaurantReview.builder()
+                        .userId(userId)
+                        .restaurantId(restaurantId)
+                        .score(RestaurantReviewScore.values()[reviewRequestDto.getScore()])
+                        .comment(reviewRequestDto.getComment())
+                        .build()));
+    }
+
+    @Transactional
+    public RestaurantReviewResponseDto updateRestaurantReview(
+            Long userId, Long reviewId, RestaurantReviewRequestDto reviewRequestDto) {
+
+        checkUserExists(userId);
+
+        RestaurantReview getRestaurantReview =
+                reviewRepository.findById(reviewId)
+                        .orElseThrow(ReviewNotFoundException::new);
+        getRestaurantReview.changeScoreAndComment(
+                reviewRequestDto.getScore(), reviewRequestDto.getComment());
+
+        return toReviewResponseDto(getRestaurantReview);
+    }
+
+    @Transactional
+    public void deleteRestaurantReview(Long userId, Long reviewId) {
+
+        checkUserExists(userId);
+
+        if (!reviewRepository.existsById(reviewId)) {
+            throw new ReviewNotFoundException();
+        }
+
+        reviewRepository.deleteById(reviewId);
+    }
```

Comment:
제 생각에는 `existsById`, `findById` 를  `findById` 하나도 처리해도 될 것 같아요!
```suggestion
    public RestaurantReviewResponseDto updateRestaurantReview(
            Long userId, Long reviewId, RestaurantReviewRequestDto reviewRequestDto) {
        checkUserExists(userId);

        RestaurantReview getRestaurantReview = findReview(reviewId);
        getRestaurantReview.changeScoreAndComment(
                reviewRequestDto.getScore(), reviewRequestDto.getComment());

        return toReviewResponseDto(getRestaurantReview);
    }

    @Transactional
    public void deleteRestaurantReview(Long userId, Long reviewId) {
        checkUserExists(userId);
        findReview(reviewId);

        reviewRepository.deleteById(reviewId);
    }
    
    private RestaurantReview findReview(Long id) {
        return reviewRepository.findById(id)
                               .orElseThrow(ReviewNotFoundException::new);
    }
```

Reply:
이부분은 findById와 existsById의 목적이 다르고, 크게 차이는 나지 않겠지만 엔티티를 데이터베이스에서 가져오냐와, 조회만 하느냐의 속도차이가 있을수 있다고 생각하여 따로 분리하여 사용하였습니다!